src/modules/keyboardmanager/common/KeyboardManagerState.cpp

```diff
@@ -278,4 +278,63 @@ bool KeyboardManagerState::DetectShortcutUIBackend(LowlevelKeyboardEvent* data)
     }
 
     return false;
+}
+
+// Save the updated configuration.
+bool KeyboardManagerState::SaveConfigToFile()
+{
+    bool result = true;
+    json::JsonObject configJson;
+    json::JsonArray remapKeysArray;
+    json::JsonArray remapShortcutsArray;
+    std::unique_lock<std::mutex> lockSingleKeyReMap(singleKeyReMap_mutex);
+    for (const auto &it : singleKeyReMap)
+    {
+        json::JsonObject keys;
+        keys.SetNamedValue(KeyboardManagerConstants::OriginalKeysSettingName, json::value(winrt::to_hstring((unsigned int)it.first)));
+        keys.SetNamedValue(KeyboardManagerConstants::NewRemapKeysSettingName, json::value(winrt::to_hstring((unsigned int)it.second)));
+
+        remapKeysArray.Append(keys);
+    }
+    lockSingleKeyReMap.unlock();
+
+    std::unique_lock<std::mutex> lockOsLevelShortcutReMap(osLevelShortcutReMap_mutex);
+    for (const auto &it : osLevelShortcutReMap)
+    {
+        json::JsonObject keys;
+        keys.SetNamedValue(KeyboardManagerConstants::OriginalKeysSettingName, json::value(it.first.ToHstringVK()));
+        keys.SetNamedValue(KeyboardManagerConstants::NewRemapKeysSettingName, json::value(it.second.targetShortcut.ToHstringVK()));
+
+        remapShortcutsArray.Append(keys);
+    }
+    lockOsLevelShortcutReMap.unlock();
+
+    configJson.SetNamedValue(KeyboardManagerConstants::RemapKeysSettingName, remapKeysArray);
+    configJson.SetNamedValue(KeyboardManagerConstants::RemapShortcutsSettingName, remapShortcutsArray);

```

Comment:
I think we have to add a subkey "global" with the remapshortcut array and then add that to the remapShortcuts key. If we don't do that then when we add in support for app specific shortcuts it would be a breaking change.